models/User.go

```diff
@@ -2095,6 +2140,50 @@ func GetStocksOwned(userId uint32) (map[uint32]int64, error) {
 	return stocksOwned, nil
 }
 
+func GetReservedStocksOwned(userId uint32) (map[uint32]int64, error) {
+	var l = logger.WithFields(logrus.Fields{
+		"method": "GetReservedStocksOwned",
+		"userId": userId,
+	})
+
+	l.Info("GetReservedStocksOwned requested")
+
+	l.Debugf("Acquiring lock on user")
+
+	ch, _, err := getUserExclusively(userId)
+	if err != nil {
+		l.Errorf("Errored: %+v", err)
+		return nil, err
+	}
+	l.Debugf("Acquired")
+	defer func() {
+		close(ch)
+		l.Debugf("Released lock on user")
+	}()
+
+	db := getDB()
+
+	sql := "SELECT SUM(stockQuantity), SUM(stockQuantityFulFilled), stockId FROM `Bids` WHERE `userId` = ? AND isClosed = 0 GROUP BY `stockId`"
```

Comment:
Aren't you supposed to make this query from the Asks table? It's Ask orders that cause stocks to be reserved. 

@PaiAkshay998 Check if this query will work.

Reply:
I guess this should be right. Whatever is not fulfilled is reserved at this point. So this should be fine.

Reply:
No, but the query is made on the wrong table no. Shouldn't it be from the `Asks` table? Not the Bids table

Reply:
When you buy an order, it gets stored in Bids table, it has ``stockQuantity`` and ``stockQuantityFulFilled``, difference of the fields basically gives the stocks that user has reserved(he paid cash but he did not get).

Reply:
The user has only reserved stocks if he places a Sell order, hence you should be checking the Asks table.